### PR TITLE
Add initial docs for Serverless Span Autolinking 

### PR DESCRIPTION
**Pull request by sumedham**
### What does this PR do? What is the motivation?
PR adds documentation for Span Auto-linking, a new feautre that the Serverless team is going to launch at re:invent.

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/serverless/aws_lambda/distributed_tracing.md [hunk 1]
```diff
@@ -108,15 +110,24 @@ For .NET serverless applications, Datadog recommends [installing Datadog's traci
 
 Learn more about [tracing through .NET Azure serverless applications][15].
 
-### Hybrid environments
+## Span Auto-linking
+Datadog can automatically detect linked spans when certain parts of your request cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These will appear in the Span Links tab in the Datadog Trace UI. 
+
+**Note:** Span auto-linking may not work if you are only ingesting a sample of your traces because the linked traces might be dropped before ingestion. To improve your chances of seeing auto-linked spans, increase your sample rate.
```
Comment (apiarian-datadog) [inline content/en/serverless/aws_lambda/distributed_tracing.md:116]: There is also the matter of not having the upstream span pointers if the object or item was created without them.

File: content/en/serverless/aws_lambda/distributed_tracing.md [hunk 2]
```diff
@@ -108,15 +110,24 @@ For .NET serverless applications, Datadog recommends [installing Datadog's traci
 
 Learn more about [tracing through .NET Azure serverless applications][15].
 
-### Hybrid environments
+## Span Auto-linking
+Datadog can automatically detect linked spans when certain parts of your request cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These will appear in the Span Links tab in the Datadog Trace UI. 
+
+**Note:** Span auto-linking may not work if you are only ingesting a sample of your traces because the linked traces might be dropped before ingestion. To improve your chances of seeing auto-linked spans, increase your sample rate. 
+
+If you are viewing the request that originated before the change event and the linked trace is ingested, you will be able to see
```
Comment (apiarian-datadog) [inline content/en/serverless/aws_lambda/distributed_tracing.md:118]: what will we se? =]

File: content/en/serverless/aws_lambda/distributed_tracing.md [hunk 3]
```diff
@@ -108,15 +110,44 @@ For .NET serverless applications, Datadog recommends [installing Datadog's traci
...
+This functionality is available for Python instrumented Lambda Functions on layer version 101 and above.
+
+### DyanmoDB Change Stream Auto-linking
+
+For [DyanmoDB Change Streams][29], Span Auto-linking supports the following operations:
+
+- `PutItem`
+- `UpdateItem`
+- `DeleteItem`
+- `BatchWriteItem`
```
Comment (apiarian-datadog) [inline content/en/serverless/aws_lambda/distributed_tracing.md:131]:
```suggestion
- `BatchWriteItem`
- `TransactWriteItems`
```

**Review by apiarian-datadog — APPROVED** [review]
left a few fixes to the supported operations. otherwise looks fine, unless we want to note that these operations are only supported in the python tracer for now and that all other tracers won't actually create span pointers.

File: content/en/serverless/aws_lambda/distributed_tracing.md [hunk 4]
```diff
@@ -108,15 +110,44 @@ For .NET serverless applications, Datadog recommends [installing Datadog's traci
 
 Learn more about [tracing through .NET Azure serverless applications][15].
 
-### Hybrid environments
+## Span Auto-linking
+Datadog automatically detects linked spans when segements of your asynchronous requests cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These span links will appear in the [Span Links tab][30] in the Datadog Trace UI.
```
Comment (brett0000FF) [inline content/en/serverless/aws_lambda/distributed_tracing.md:114]:
```suggestion
Datadog automatically detects linked spans when segments of your asynchronous requests cannot propagate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These span links appear in the [Span Links tab][30] of the Datadog Trace View. 
```

File: content/en/serverless/aws_lambda/distributed_tracing.md [hunk 5]
```diff
@@ -108,15 +110,44 @@ For .NET serverless applications, Datadog recommends [installing Datadog's traci
 
 Learn more about [tracing through .NET Azure serverless applications][15].
 
-### Hybrid environments
+## Span Auto-linking
+Datadog automatically detects linked spans when segements of your asynchronous requests cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These span links will appear in the [Span Links tab][30] in the Datadog Trace UI. 
+
+**Note:** Span auto-linking may not link traces if you are only ingesting a sample of your traces because the linked traces might be dropped before ingestion. To improve your chances of seeing auto-linked spans, increase your sample rate.
```
Comment (brett0000FF) [inline content/en/serverless/aws_lambda/distributed_tracing.md:116]:
```suggestion
**Note:** Span auto-linking may not link traces if you are only ingesting a sample of your traces, as the linked traces might be dropped before ingestion. To increase the likelihood of seeing auto-linked spans, consider increasing your sample rate.
```

File: content/en/serverless/aws_lambda/distributed_tracing.md [hunk 6]
```diff
@@ -108,15 +110,44 @@ For .NET serverless applications, Datadog recommends [installing Datadog's traci
 
 Learn more about [tracing through .NET Azure serverless applications][15].
 
-### Hybrid environments
+## Span Auto-linking
+Datadog automatically detects linked spans when segements of your asynchronous requests cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These span links will appear in the [Span Links tab][30] in the Datadog Trace UI. 
+
+**Note:** Span auto-linking may not link traces if you are only ingesting a sample of your traces because the linked traces might be dropped before ingestion. To improve your chances of seeing auto-linked spans, increase your sample rate. 
+
+If you are viewing the request that originated before the change event and the linked trace is ingested, you can the linked span as a `Backward` link.
```
Comment (brett0000FF) [inline content/en/serverless/aws_lambda/distributed_tracing.md:118]:
```suggestion
If you are viewing the request that originated before the change event and the linked trace is ingested, you can see the linked span as a `Backward` link. 
```

File: content/en/serverless/aws_lambda/distributed_tracing.md [hunk 7]
```diff
@@ -108,15 +110,44 @@ For .NET serverless applications, Datadog recommends [installing Datadog's traci
...
 
-### Hybrid environments
+## Span Auto-linking
+Datadog automatically detects linked spans when segements of your asynchronous requests cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These span links will appear in the [Span Links tab][30] in the Datadog Trace UI. 
+
+**Note:** Span auto-linking may not link traces if you are only ingesting a sample of your traces because the linked traces might be dropped before ingestion. To improve your chances of seeing auto-linked spans, increase your sample rate. 
+
+If you are viewing the request that originated before the change event and the linked trace is ingested, you can the linked span as a `Backward` link. 
+
+If you are viewing the request that originated before the change event and the linked trace is ingested, you can see the linked span as a `Forward` link.
```
Comment (brett0000FF) [inline content/en/serverless/aws_lambda/distributed_tracing.md:120]: Is this a link in the UI that says "Forward" or "Backward". Just want to make sure it's clear. Thanks!

**Review by brett0000FF — COMMENTED** [review]
Just a couple minor fixes and a clarifying question. Thanks!

File: content/en/serverless/aws_lambda/distributed_tracing.md (hunk 7 above)
Comment (sumedham) [reply content/en/serverless/aws_lambda/distributed_tracing.md:120]: yeah it's a tag, it looks like this - 

image

**Review by brett0000FF — APPROVED** [review]

**Comment by brett0000FF** [comment]
@sumedham - Heads up that you can comment `/merge` on this PR to merge it when you are ready!

**Pull request merged**